tools/structure/src/main.py

```diff
@@ -86,6 +86,10 @@ def run(
         self.stream_log("Indexing document")
         usage_kwargs: dict[Any, Any] = dict()
         usage_kwargs[SettingsKeys.RUN_ID] = run_id
+        usage_kwargs[SettingsKeys.FILE_NAME] = (
```

Comment:
If necessary, we can add it as a dataclass maybe to keep track of all possible keys this can take -is it documented anywhere @johnyrahul @Deepak-Kesavan ?

Reply:
Will take this as a separate activity.